### ISSUE TITLE
Can't get it to work at all on multiple computers

### ISSUE THREAD

**Issue by Genus-learner** [darkarp/chromepass#35]
I tried the steps on different windows 10 computers and cannot get it to work. There is always some sort of error. I don't know anything about programming so I don't know the issue.
I currently don't have any screenshots of the error because I uninstalled/deleted everything related to it. I also couldn't get your malkit to work.

I am currently downloading a Windows 10 VM in a last ditch effort.
I just gotta ask: Does this only work on a VM?

**Comment by Genus-learner** [comment]
I downloaded a VM on two computers
on one of them it gets stuck here during setup
![image](https://user-images.githubusercontent.com/83723816/117222030-21fd8e00-add0-11eb-9198-48aa6230a6f7.png)

On the other machine I see this error
![image](https://user-images.githubusercontent.com/83723816/117222053-2fb31380-add0-11eb-8ad7-2211a8979ba7.png)
![image](https://user-images.githubusercontent.com/83723816/117222085-45283d80-add0-11eb-9103-f05e02b4e9bb.png)

if i try to run the create.py it says this
![image](https://user-images.githubusercontent.com/83723816/117222196-8587bb80-add0-11eb-8ebf-6d50b3e68b73.png)
and becomes an endless loop

**Comment by Genus-learner** [comment]
tried installing depedencies manually

Here's an error:
![image](https://user-images.githubusercontent.com/83723816/117224006-8fabb900-add4-11eb-99d4-53d8eecc301c.png)

**Comment by Genus-learner** [comment]
after a lot of work i got it to run and create the server/client exe
![image](https://user-images.githubusercontent.com/83723816/117231952-ea014580-ade5-11eb-9610-d66e7e0ba485.png)
but there is no json file created with passwords

**Comment by Genus-learner** [comment]
I seem to have managed to get it to work with the python server offline
but now I have trouble port forwarding?
I followed the guide you sent and made a port, I put my IP address and port when creating the server but the client doesnt send anything from other computers

**Comment by Genus-learner** [comment]
Update:
I now have it sending info from a different computer
but it wont do it from another network
which defeats the purpose

**Comment by Genus-learner** [comment]
I might have gotten it working by using my public IP
here's to hoping

**Comment by darkarp** [comment]
Hi, I'm aware of this issue, regarding the dependency installation and detection. It will be fixed on next update, 
I'm just incredibly busy at the moment I'm sorry. That problem lies in the fact that currently I'm using `Chocolatey` to install `vcbuildtools`, which is a dependency, but unfortunately that `choco` package hasn't been updated in a long time and is not working anymore. 

In the next update I'll include the dependencies myself, installing the `vcbuildtools` directly.

As for the other problem, it should work on any network provided correct port forward setup. I haven't tested in such an environment, but considering that if port forwarding is setup correctly the network behaves exactly the same way as it does in local area networks.
 I'll just give a quick recap in case they weren't properly configured:
 - You have to use your public IP address in the program so it can listen for connections from outside your LAN.
 - In your router you must have a port forwarding rule that forwards TCP traffic from the port the program uses (default 80), into your computer (LAN ip address).
 - You can check if correct port forwarding has happened by visiting [this website](https://yougetsignal.com/tools/open-ports/) and typing your public IP and the port, it will tell you whether it's open or closed. You want it to be open.
 - That's basically it. You should now be able to open the client on any computer and the server should catch it.

**Comment by Genus-learner** [comment]
Yeah I slowly worked through all the dependency issues as you see in the thread.
Something I didnt mention is that after manually installing all the listed dependencies it still said it was missing them. So I went into the source code and commented out the line that prints that message and when I ran the code it instead told me what was actually missing. "No module named PyInstaller". So i installed that and it finally ran clean.
After figuring out port forwarding I kept using the wrong IP when building the server. 
but eventually
I got it working with my last post.

I think the something like using the public IP should be made clear in the instructions for idiots like myself.
Thanks for your response though I wasn't sure if this page was still active.

I think its working now so I'll close this issue. Have a good one. Hopefully something in my thread will help anyone having issues in the future.
